### DISCUSSION TITLE
无法通过配置模板与option将menu里面option所设定的值同步到config.h

### DISCUSSION THREAD

**Discussion by chenlilong84** [xmake-io/xmake#3765] [Q&A] [answered]
目前我正在学习如何通过配置模板以及xmake f --menu通过设置option去讲项目的配置存到config.h里面，但是在使用过程中，发现没有成功。
下面是我的xmake.lua demo code:
`target("test")
    set_kind("binary")
    add_files("src/*.c")
    -- set the auto-generated config.h
--    set_configdir("$(buildir)/$(plat)/$(arch)/$(mode)")
    add_configfiles("config.h.in", {filename = "myconfig.h"})  -- rename output config name
    add_options("demo1")

option("demo1")
    set_default("123")
    set_showmenu(true)
    set_category("demo")
    set_configvar("CONFIG_DEMO1_VALUE", "123")	--get_config("demo1")
	set_description("Enable or disable demo1")
option_end()
`
config.h.in内容是：
`#define CONFIG_DEMO1_VALUE "${CONFIG_DEMO1_VALUE}"`

当我尝试使用menu去修改option demo1的时候，始终无法同步到产生的config.h里面，产生的config.h里面的CONFIG_DEMO1_VALUE值一直是123.  还请帮忙厘清下，是不是我使用上有什么问题，谢谢！

**Comment by waruqi** [comment, marked as answer]
> set_configvar("CONFIG_DEMO1_VALUE", "123") 

你写死了，当然一直是 123。。得在 after_check 脚本里面，通过 `option:set("configvar", "CONFIG_DEMO1_VALUE", option:value())` 设置

https://xmake.io/#/manual/configuration_option?id=optionafter_check

**Reply by chenlilong84** [reply]
Thanks a lot.
